controllers/application/deployment.go

```diff
@@ -44,7 +46,11 @@ func (r *ApplicationReconciler) reconcileDeployment(ctx context.Context, applica
 
 		r.SetLabelsFromApplication(ctx, &deployment, *application)
 
-		deployment.Spec.Template.ObjectMeta.Annotations = map[string]string{"prometheus.io/scrape": "true"}
+		deployment.ObjectMeta.Annotations = util.CommonAnnotations
+		deployment.Spec.Template.ObjectMeta.Annotations = map[string]string{
+			"argocd.argoproj.io/sync-options": "Prune=false",
+			"prometheus.io/scrape":            "true",
+		}
```

Comment:
Since this might be common pattern going forward (common annotation + a special case) I propose the following
```suggestion
		deployment.Spec.Template.ObjectMeta.Annotations = util.CommonAnnotationsPlus(
			"prometheus.io/scrape", "true",
		)
```

And adding the following function in `pkg/util/helperfunctions.go`:
```go
func CommonAnnotationsPlus(kvs ...string) map[string]string {
	if len(kvs)%2 != 0 {
		panic(fmt.Sprintf("both key and value must be supplied, got %s", kvs))
	}

	m := CommonAnnotations
	for i := 0; i < len(kvs); i = i + 2 {
		m[kvs[i]] = kvs[i+1]
	}

	return m
}
```